src/server/routes.pl

```diff
@@ -774,57 +774,32 @@
             ->  true
             ;   Search = []),
 
-            (   get_dict(graph_type, Posted, Graph_Type_String)
-            ->  atom_string(Graph_Type, Graph_Type_String)
-            ;   memberchk(graph_type=Graph_Type, Search)
-            ->  do_or_die(memberchk(Graph_Type, [schema, instance]),
-                          error(unknown_graph_type(Graph_Type),_))
+            (   param_value_search_or_json(Search, JSON, graph_type, graph, Graph_Type)
```

Comment:
Fair number of choice points set up here. This increases the stack size and can create surprising behaviour if we back track over these to find the default solution which is probably not what is meant.  (this could be controlled with a cut at the end of the parameter selections but that is non-local and confusing) 

It is better to have a default value version of `param_value_search_or_json` which takes another arg and uses if-then-else. 

i.e.
```prolog
param_value_search_or_json(Search, JSON, Arg, Type, Default, Value) :- 
    (    param_value_search_or_json(Search, JSON, Arg, Type, Value) 
    ->  true
    ;    Value = Default
    ).
```

Reply:
Great! I just have to remember that `->/2` implies a cut and that `(... ; ...)` without `->/2` implies there may be unexpected backtracking.